dbms/src/Storages/StorageFile.cpp

```diff
@@ -143,8 +201,8 @@ class StorageFileBlockInputStream : public IBlockInputStream
         else
         {
             shared_lock = std::shared_lock(storage.rwlock);
-
-            read_buf = std::make_unique<ReadBufferFromFile>(storage.path);
+            std::string path_to_read = storage.path_with_globs ? storage.matched_paths[num_of_path] : storage.path;
```

Comment:
There are some other file-related fields in StorageFile, such as `storage.table_fd_was_used` or `storage.rwlock`. Should they be different for each file?

Reply:
No, they don't

Reply:
fd doesn't used in this case, and rwlock is common because path with globs might be only readonly now